packages/runtime/container-runtime/src/summaryManagerUtils.ts

```diff
@@ -0,0 +1,200 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IEvent } from "@fluidframework/common-definitions";
+import { assert, TypedEventEmitter } from "@fluidframework/common-utils";
+import { IQuorum, ISequencedClient } from "@fluidframework/protocol-definitions";
+
+export const summarizerClientType = "summarizer";
+
+export interface ITrackedClient {
+    readonly clientId: string;
+    readonly sequenceNumber: number;
+}
+
+interface ILinkedClient extends ITrackedClient {
+    prevClient: ILinkedClient | undefined;
+    nextClient: ILinkedClient | undefined;
+}
+
+interface IFullClient extends ILinkedClient {
+    readonly isSummarizer: boolean;
+}
+
+interface IOrderedClientsEvents extends IEvent {
+    (event: "currentChange", listener: (client: ITrackedClient | undefined) => void);
```

Comment:
some comments on what these mean would be nice. not immediatly obvious what currentChange is